### PR TITLE
New analysis task added to PWGJE: calculating Jet Charge

### PR DESCRIPTION
**Pull request by JColburn-CERN**
Added a new analysis task AliAnalysisTaskJetCharge which calculates the jet charge distributions for either pp or PbPb.
This has been added such that it can be run on the LEGO trains system.
Written by me, J Samuel Colburn. Originally uploaded on the 28th of October.

### CODE REVIEW

File: PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeTemplates.h [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    {
+    kIsGhost = IsGhost;
+    fCharge = Charge;
+    fPartonPdg = PartonPdg;
+    }
+
+  Int_t getCharge() const{
+    return fCharge;
+  }
+};
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeTemplates.h:42]: This is a really generic name to put into the global namespace

**Review by raymondEhlers — CHANGES_REQUESTED** [review]
Hi,

You need to include these files in the CMakeLists and the LinkDef from them to actually be usable in AliPhysics. I didn't review your code in detail, but made one comment on one of your classes - the name is quite generic and may conflict with others.

Best regards,
Raymond

**Comment by JColburn-CERN** [comment]
Hi Raymond,
Thank you for your feedback, This is my first time adding to the repository so I was unaware of those steps so I have now added the files to the cmake list and linkdef.
I have renamed AliAanalysisTaskJetChargeTemplates to AliAnalysisTaskJetChargeFlavourTemplates to make the name less generic. Should I also remane AliAnalysisJetTaskJetCharge to add spesificity?

Thank you for your help
Best regards
Sammy

**Comment by raymondEhlers** [comment]
Hi Sammy,

Thanks for the fixes. I wasn't actually referring to you task name - it's just fine. My concern was about the class called `MyInfo`, which is why I highlighted it in my review.

Best,
Raymond

**Comment by JColburn-CERN** [comment]
Hi Raymond,
Ah thank you for the clarification that makes more sense,
Also thank you for bringing it to my attention since MyInfo was simply a holdover from an older piece of code and no longer being used so I simply removed it.
Cheers
Sammy

**Comment by raymondEhlers** [comment]
Hi Sammy,

Thanks for the fixes. You should fix the issues identified by the testing, as well as squash your commits. Also, I don't see a LinkDef entry. Lastly, since you explicitly include fastjet, you need to move to the fastjet section of the CMakeLists. Same for the LinkDef.

Best,
Raymond

**Comment by JColburn-CERN** [comment]
HI Reymond,
Thanks for running the tests. I believe I have fixed the issues that were present in the tests as well as having moved the entry on the CMakeList and added the Linkdef
Thank you for the help,
Sammy

**Comment by raymondEhlers** [comment]
+1 Looks good - thanks Sammy

**Comment by raymondEhlers** [comment]
+1

**Comment by raymondEhlers** [comment]
+1

**Pull request merged**